library/core/src/fmt/mod.rs

```diff
@@ -2281,16 +2282,51 @@ impl<T: ?Sized> Pointer for &mut T {
 
 // Implementation of Display/Debug for various core types
 
+/// A local trait for pointer Debug impl so that we can have
+/// min_specialization-compliant specializations for two types of fat ptrs.
+trait PtrMetadataFmt {
+    fn fmt_ptr(&self, ptr: *const (), f: &mut Formatter<'_>) -> Result;
+}
+
+// Regular pointer / default impl
+impl<T> PtrMetadataFmt for T {
+    #[inline]
+    default fn fmt_ptr(&self, ptr: *const (), f: &mut Formatter<'_>) -> Result {
+        Pointer::fmt(&ptr, f)
+    }
+}
+
+// Pointer + length
+impl PtrMetadataFmt for usize {
+    #[inline]
+    fn fmt_ptr(&self, ptr: *const (), f: &mut Formatter<'_>) -> Result {
+        write!(f, "({:p}, {})", ptr, *self)
```

Comment:
I wonder how silly something like `*[_] { data: 0x..., len: 123 }` would be. (and have `meta` instead of `len` in the general case, I guess?)

Reply:
I'm not sure I understand, what's the `*[_]` part?